docs/grammar/Solidity.g4

```diff
@@ -0,0 +1,489 @@
+    Function LParen (arguments=parameterList)? RParen
+    (visibility | stateMutability)*
+    (Returns LParen returnParameters=parameterList RParen)?;
+
+/**
+ * The declaration of a single variable.
+ */
+variableDeclaration: type=typeName location=storageLocation? name=identifier;
+storageLocation: Memory | Storage | Calldata;
+
+/**
+ * Complex expression.
+ * Can be an index access, an index range access, a member access, a function call (with optional function call options),
+ * a type conversion, an unary or binary expression, a comparison or assignment, a ternary expression,
+ * a new-expression (i.e. a contract creation or the allocation of a dynamic memory array),
+ * a tuple, an inline array or a primary expression (i.e. an identifier, literal or type name).
+ */
+expression:
+	expression LBrack index=expression? RBrack # IndexAccess
+	| expression LBrack start=expression? Colon end=expression? RBrack # IndexRangeAccess
+	| expression Period (identifier | Address) # MemberAccess
+	| expression LBrace (namedArgument (Comma namedArgument)*)? RBrace # FunctionCallOptions
+	| expression callArgumentList # FunctionCall
+	| Payable callArgumentList # PayableConversion
+	| Type LParen typeName RParen # MetaType
+	| (Inc | Dec | Not | BitNot | Delete | Sub) expression # UnaryPrefixOperation
+	| expression (Inc | Dec) # UnarySuffixOperation
+	|<assoc=right> expression Exp expression # ExpOperation
+	| expression (Mul | Div | Mod) expression # MulDivModOperation
+	| expression (Add | Sub) expression # AddSubOperation
+	| expression (Shl | Sar | Shr) expression # ShiftOperation
+	| expression BitAnd expression # BitAndOperation
+	| expression BitXor expression # BitXorOperation
+	| expression BitOr expression # BitOrOperation
+	| expression (LessThan | GreaterThan | LessThanOrEqual | GreaterThanOrEqual) expression # OrderComparison
+	| expression (Equal | NotEqual) expression # EqualityComparison
+	| expression And expression # AndOperation
+	| expression Or expression # OrOperation
+	|<assoc=right> expression Conditional expression Colon expression # Conditional
+	|<assoc=right> expression assignOp expression # Assignment
+	| New typeName # NewExpression
+    | tupleExpression # Tuple
+    | inlineArrayExpression # InlineArray
+ 	| (
+		identifier
+		| literal
+		| elementaryTypeName[false]
+		| userDefinedTypeName
+	  ) # PrimaryExpression
+;
+
+//@doc:inline
+assignOp: Assign | AssignBitOr | AssignBitXor | AssignBitAnd | AssignShl | AssignSar | AssignShr | AssignAdd | AssignSub | AssignMul | AssignDiv | AssignMod;
+tupleExpression: LParen (expression? ( Comma expression?)* ) RParen;
+/**
+ * An inline array expression denotes a statically sized array of the common type of the contained expressions.
+ */
+inlineArrayExpression: LBrack (expression ( Comma expression)* ) RBrack;
+
+/**
+ * Besides regular non-keyword Identifiers, the 'from' keyword can also occur as identifier outside of import statements.
+ */
+identifier: Identifier | From;
```

Comment:
Kinda hacky innit?

Reply:
Tell that to our parser :-). See https://github.com/ethereum/solidity/issues/9621